### ISSUE TITLE
Frida-compile module resolution problems in Ubuntu 20.04

### ISSUE THREAD

**Issue by oriori1703** [frida/frida-core#458]
There seems to be a module resolution problem in Ubuntu older version, which doesn't occur in newer Ubuntu versions.

I encountered, and can reproduce the problem, in Ubuntu 20.04, but can't in Ubuntu 22.04.
I haven't had the time to check exactly which ubuntu version in the middle fixes the problem.

Whenever I import anything in my `index.ts`, and then try to compile it using `frida-compile agent/index.ts` I get an error similar to this:
```
agent/index.ts:1:21 - error TS2307: Cannot find module './logger.js' or its corresponding type declarations.
compilation failed
```

The same thing happens when I try to compile using frida-python directly, using `frida.Compiler`.

To help reproduce this I create a minimal dockerfile that encounters this error:
```dockerfile
FROM ubuntu:focal
RUN apt-get update && apt-get install -y python3 python3-pip
RUN python3 -m pip install frida-tools

RUN frida-create -t agent -o compile-test && sed -i 's/"\.\/logger"/"\.\/logger.js"/' compile-test/agent/index.ts
WORKDIR compile-test
CMD ["frida-compile", "agent/index.ts"]
``` 

If you change `ubuntu:focal` to `ubuntu:jammy` you can see that the problem doesn't occur.

All my testing was done using both `frida==16.0.9` and `frida==16.0.10` .

I hope this helps :)

**Comment by milahu** [comment]
1. this belongs to https://github.com/frida/frida-compile
2. debian/ubuntu packages are always old, and backward-compatibility should have low priority
3. https://duckduckgo.com/?q=old+node+version+error+TS2307+Cannot+find+module+or+its+corresponding+type+declarations

**Comment by oriori1703** [comment]
Sorry for the confusion. I do think the issue belongs to this repo, because the issue only occurs when using frida-compile from frida-tools.
It doesn't occur when using the npm version.

If I understand everything correctly, the `frida-compile`from `frida-tools` uses `frida-python`'s frida.Compiler. This in turn injects frida into the same python process, and runs then npm version of `frida-compile` in Frida's injected v8 engine.

Therefore I concluded that this is not an issue related to:
1. The npm `frida-compile` because that actually works for me.
2. The node version, because node is not used, and is not even installed in the docker example.
3. A problem with the project config / module resolution because it works with the npm version of `frida-compile`

**Comment by oriori1703** [comment]
My guess is that it is a problem in `patchCompilerHost` at [agent-core.ts](https://github.com/frida/frida-core/blob/main/src/compiler/agent-core.ts), but I didn't have enough time to compile frida-core, and debug it my self.
Hopefully I will have time during the weekend if nobody else solves this before then :)

**Comment by milahu** [comment]
> the issue only occurs when using frida-compile from frida-tools. It doesn't occur when using the npm version.

maybe frida-tools uses an old version of frida-compile?

at least frida-gum does, see also https://github.com/frida/frida-gum/issues/724#issuecomment-1438909906

**Comment by oriori1703** [comment]
It seems to be using the latest `frida-compile`:
https://github.com/frida/frida-core/blob/af07acd447d9084e9a8dab4e7499e843a50b229f/src/compiler/package.json#L10-L14

**Comment by oleavr** [comment]
Thanks for reporting! This was finally fixed in [16.0.14](https://frida.re/news/2023/04/18/frida-16-0-14-released/). The [fix](https://github.com/nowsecure/frida-fs/commit/a1d55bc9b309311fc2ac16864cbe414cb4cacaaf) happened in frida-fs.

**Comment by oriori1703** [comment]
Thanks 😊
